webrtc/README.md

```diff
@@ -0,0 +1,317 @@
+# WebRTC
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Working Draft | Active |                 |
+
+Authors: [@mxinden]
+
+Interest Group: [@marten-seemann]
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+**Table of Contents**
+
+- [WebRTC](#webrtc)
+    - [Motivation](#motivation)
+    - [Requirements](#requirements)
+    - [Addressing](#addressing)
+    - [Connection Establishment](#connection-establishment)
+        - [Browser to public Server](#browser-to-public-server)
+            - [Open Questions](#open-questions)
+        - [Browser to Browser](#browser-to-browser)
+            - [Open Questions](#open-questions-1)
+    - [Connection Security](#connection-security)
+        - [Open Questions](#open-questions-2)
+    - [Multiplexing](#multiplexing)
+    - [General Open Questions](#general-open-questions)
+    - [Previous, ongoing and related work](#previous-ongoing-and-related-work)
+- [FAQ](#faq)
+
+<!-- markdown-toc end -->
+
+
+## Motivation
+
+1. **No need for valid TLS certificates.** Enable browsers to connect to public
+   server nodes without those server nodes providing a TLS certificate within
+   the browsers trustchain. Note that we can not do this today with our
+   Websocket transport.
+
+2. **Hole punching in the browser**: Enable two browsers or a browser and a
+   non-public server node to connect.
+
+## Requirements
+
+- Loading a remote nodes certificate into ones browser trust-store is not an
+  option, i.e. doesn't scale.
+
+- No dependency on central STUN and/or TURN servers.
+
+## Addressing
+
+WebRTC multiaddresses are composed of an IP and UDP address component, followed
+by `/webrtc` and a multihash of the certificate that the node uses.
```

Comment:
In https://github.com/multiformats/multiaddr/pull/130#issuecomment-1186540999 @tomaka suggests to use `/webrtc-direct` instead of `/webrtc`. I think this is a better place for this discussion to take place.

> Why `/webrtc-direct` instead of `/webrtc`? Because we know that we will add alternative WebRTC protocols in the future. I like the fact that `/webrtc-direct` conveys the idea that this is one opinionated way of connecting through WebRTC, rather than _the_ WebRTC protocol.

Agreed that we might add other flavors (e.g. always double encrypting via nested Noise), though I would expect that the flavor we are adding here (this specification) will be the most used one, thus deserving the short name.

To the best of my knowledge, we don't need a separate protocol name for the browser-to-browser use-case. I.e. we can use the same `/webrtc` protocol name. Though, given that we haven't yet designed it, I might be missing something.

What do folks think? @tomaka would you expect as well that this flavor will become the most prominent one and thus deserves the shorter name?

Reply:
A more specific name is a good thing, given that we already have [webrtc-star](https://www.npmjs.com/package/@libp2p/webrtc-star) and [webrtc-direct](https://www.npmjs.com/package/@libp2p/webrtc-direct) transports.

These use the multiaddr formats `/ip4/.../tcp/.../wss/p2p-webrtc-star/p2p/QmFoo` and `/ip4/.../tcp/.../http/p2p-webrtc-direct/QmFoo` respectively.

Reply:
I am not going to push for this, i.e. I don't intend to change the protocol name. In case folks do feel strongly about this, please speak up.